src/SourceBuild/Arcade/eng/common/templates/job/source-build-build-tarball.yml

```
@@ -68,7 +68,7 @@ jobs:
   - ${{ if ne(parameters.excludeSdkContentTests, 'true') }}:
     - download: ${{ parameters.installerBuildResourceId }}
       artifact: BlobArtifacts
-      patterns: '**/dotnet-sdk-!(*-*)-linux-${{ parameters.architecture }}.tar.gz'
+      patterns: '**/dotnet-sdk-+([0-9]).+([0-9]).+([0-9])?(-preview*)?(-rc*)-linux-${{ parameters.architecture }}.tar.gz'
```

Consider using `|` to be more precise.

```suggestion
      patterns: '**/dotnet-sdk-+([0-9]).+([0-9]).+([0-9])?(-(preview|rc)*)?-linux-${{ parameters.architecture }}.tar.gz'
```